spark/src/main/scala/org/apache/spark/sql/delta/DeltaOperations.scala

```diff
@@ -482,10 +484,12 @@ object DeltaOperations {
   /** Recorded when optimizing the table. */
   case class Optimize(
       predicate: Seq[Expression],
-      zOrderBy: Seq[String] = Seq.empty
+      zOrderBy: Seq[String] = Seq.empty,
+      auto: Boolean = false
```

Comment:
Think `autoCompact` would be more meaningful

Reply:
re https://github.com/delta-io/delta/pull/2414#issuecomment-1883922342, perhaps we can address post merge